### PR TITLE
make reqwest optional for azure_core

### PR DESCRIPTION
**Pull request by ctaggart**
For #735, it adds a check for `cargo check -p azure_core --no-default-features` which does not include `enable_reqwest`. Note, that these legacy errors can't be removed until  #771.

### CODE REVIEW

File: sdk/core/src/options.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ use std::time::Duration;
 ///     .retry(RetryOptions::default().max_retries(10u32))
 ///     .telemetry(TelemetryOptions::default().application_id("my-application"));
 /// ```
-#[derive(Clone, Debug, Default)]
```
Comment (rylev) [inline sdk/core/src/options.rs:18]: Why was this removed? I think what there is now is equivalent only more verbose.

File: sdk/core/src/response.rs [hunk 2]
```diff
@@ -13,6 +13,7 @@ pub(crate) struct ResponseBuilder {
 }
 
 impl ResponseBuilder {
+    #[allow(dead_code)]
```
Comment (rylev) [inline sdk/core/src/response.rs:16]: This maybe points to that this module should only be built if the reqwest feature is turned on.

Comment (ctaggart) [reply sdk/core/src/response.rs]: I updated to use cfg.

File: sdk/core/src/options.rs (hunk 1 above)
Comment (rylev) [reply sdk/core/src/options.rs:18]: Ah, can you use [`cfg_attr`](https://stackoverflow.com/questions/42551113/is-it-possible-to-conditionally-enable-an-attribute-like-derive) for this?
